.github/workflows/rerun_workflow_on_push.yml

```diff
@@ -0,0 +1,79 @@
+---
+name: Rerun affected workflows in long lived branches on push
+
+on:
+  push:
+    branches:
+      - main
+      - candlepin-*-HOTFIX
+
+jobs:
+  rerun-workflows:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Mask secrets
+        shell: bash
+        run: |
+          echo "::add-mask::${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Get PR info
+        uses: 8BitJonny/gh-get-current-pr@fa2160116055f6fe10376375892b255f8b2ed8f6
+        id: PR
+        with:
+          github-token: ${{ github.token }}
+
+      - name: Rerun workflows
+        if: ${{ !contains(steps.PR.outputs.pr_labels, 'skip CI') }}
+        shell: bash
+        run: |
+          base_branch="${GITHUB_REF#refs/heads/}"
+          
+          # List affected PRs exclude dependabots, weblate and draft PRs
+          pr_head_shas=$(gh api \
+          -H "Accept: application/vnd.github+json" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          /repos/candlepin/candlepin/pulls | \
+          jq --arg base "$base_branch" -r '.[] | select(.base.ref == $base and .draft == false and .user.login != "weblate" and .user.login != "dependabot[bot]") | .head.sha')
+
+          # List runner ids from PRs(only for pr_verification.yml)
+          for head_sha in $pr_head_shas; do
+            runner_id=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            /repos/candlepin/candlepin/actions/workflows/pr_verification.yml/runs?head_sha=$head_sha | \
+            jq -r '.workflow_runs[] | .id')
+            runner_ids+=($runner_id)
+          done
+          
+          # Cancel running workflows
+          for id in "${runner_ids[@]}"; do
+            # Get status of running workflow
+            status=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            /repos/candlepin/candlepin/actions/runs/$id | \
+            jq -r '.status')
+          
+            # Cancel workflow if it in progress or in queue
+            if [ "$status" = "in_progress" ] || [ "$status" = "queued" ]; then
+              gh api \
+              --method POST \
+              -H "Accept: application/vnd.github+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              /repos/candlepin/candlepin/actions/runs/$id/cancel
+            fi
+          done
+          
+          # We need to sleep here because canceling workflows take some time
+          sleep 15
```

Comment:
Sleeping usually leads to intermittent failures. what happens if one of the workflows takes longer than 15 to get cancelled? Perhaps a do-while loop over the runner_ids where you check that all of them have been cancelled is better here?

Reply:
Or another approach would be to keep the sleep here, but in the next step (rerun workflows), you use some kind of try/catch, around the API call, so that if one call fails, the loop continues to call all the rest of them, instead of the whole workflow failing?

Reply:
done